### PR TITLE
Fix array2string for specific mat type

### PR DESCRIPTION
**Pull request by birm**
re: #7687, #7705

### CODE REVIEW

**Comment by seberg** [comment]
Just to note, there is usually no need to open a new PR, you can just force push to update the old one. It is however best if you get used to not using the master branch (on your clone), but instead create a new branch first and push that back to github. This saves you trouble with any future work, and in case of necessary cleanup, because then you can keep your master branch in sync with numpy's master branch.

**Comment by charris** [comment]
Work in a branch, not master. Also see `doc/source/dev/gitwash/development_workflow.rst` for formatting of commit messages. Commits also need squashing and tests are failing.

File: numpy/core/arrayprint.py [hunk 1]
```diff
@@ -434,6 +434,10 @@ def array2string(a, max_line_width=None, precision=None,
 
     """
 
+    if (repr(type(a))=="<class 'numpy.matrixlib.defmatrix.matrix'>") and style is repr:
```
Comment (charris) [inline numpy/core/arrayprint.py]: Why not `isinstance(a, np.matrix)`?

**Comment by birm** [comment]
I'm sorry if I'm misunderstanding the tests, but isn't the test failing due to the test not expecting a "fixed" repr for mat?
`AssertionError: '\nArrays are not equal\n\n(shapes (2,), (1, 2) mismatch)\n x: array([1, 2])\n y: matrix(<bound method matrix.__str__ of matrix([[1, 2]])>)' != '\nArrays are not equal\n\n(shapes (2,), (1, 2) mismatch)\n x: array([1, 2])\n y: [repr failed for <matrix>: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()]'`

Also, I'd like to thank both of you for taking time to help me improve my workflow and commits. I hope to provide better quality help in the future!

**Comment by birm** [comment]
The conflicts have been resolved.

**Comment by birm** [comment]
To me, it looks like the other changes have resolved this problem. Anyone watching think I shouldn't just close this PR?

**Pull request closed without merging**